src/client/get.ts

```diff
@@ -159,3 +159,45 @@ export async function paginateWithCursor<Req extends Record<string, any>, Res ex
   } while (cursor !== null && cursor !== undefined);
   return out as Res;
 }
+
+/// This function is a helper for paginating using a function wrapping an API using offset instead of start
+export async function paginateWithObfuscatedCursor<Req extends Record<string, any>, Res extends Array<{}>>(
```

Comment:
iiuc, the only 2 differences is the `start vs offset` params and the `re-evaluate length calculation` - any reason not to simply use the existing `paginateWithCursor` but just provide a flag for the `re-evaluate length calculation` part? `requestParams` are already being passed as an argument to the `paginateWithCursor` function.

Reply:
I wanted to be more explicit because the `start` vs `offset` is actually more of the problem that the one with the `obfuscated cursor` is not useful / a user shouldn't provide it directly.  You would have to get it from a previous function call.

Also, I think the cursor header isn't always set, so that whole logic probably needs to be checked.

Basically there are 2 code paths on the serverside for pagination, and I'm trying to separate them in teh event we have other errors with it.